src/vtbackend/Functions.h

```diff
@@ -461,154 +465,161 @@ constexpr inline auto CaptureBufferCode = 314;
-            LS0,
-            LS1,
-
-            // ESC
-            DECALN,
-            DECBI,
-            DECFI,
-            DECKPAM,
-            DECKPNM,
-            DECRS,
-            DECSC,
-            HTS,
-            IND,
-            NEL,
-            RI,
-            RIS,
-            SCS_G0_SPECIAL,
-            SCS_G0_USASCII,
-            SCS_G1_SPECIAL,
-            SCS_G1_USASCII,
-            SS2,
-            SS3,
-
-            // CSI
-            ANSISYSSC,
-            XTCAPTURE,
-            CBT,
-            CHA,
-            CHT,
-            CNL,
-            CPL,
-            CPR,
-            CUB,
-            CUD,
-            CUF,
-            CUP,
-            CUU,
-            DA1,
-            DA2,
-            DA3,
-            DCH,
-            DECCARA,
-            DECCRA,
-            DECDC,
-            DECERA,
-            DECFRA,
-            DECIC,
-            DECSCA,
-            DECSED,
-            DECSERA,
-            DECSEL,
-            XTRESTORE,
-            XTSAVE,
-            DECPS,
-            DECRM,
-            DECRQM,
-            DECRQM_ANSI,
-            DECRQPSR,
-            DECSASD,
-            DECSCL,
-            DECSCPP,
-            DECSCUSR,
-            DECSLRM,
-            DECSM,
-            DECSNLS,
-            DECSSDT,
-            DECSTBM,
-            DECSTR,
-            DECXCPR,
-            DL,
-            ECH,
-            ED,
-            EL,
-            HPA,
-            HPR,
-            HVP,
-            ICH,
-            IL,
-            REP,
-            RM,
-            SCOSC,
-            SD,
-            SETMARK,
-            SGR,
-            SM,
-            SU,
-            TBC,
-            VPA,
-            WINMANIP,
-            XTPOPCOLORS,
-            XTPUSHCOLORS,
-            XTREPORTCOLORS,
-            XTSHIFTESCAPE,
-            XTSMGRAPHICS,
-            XTVERSION,
-
-            // DCS
-            STP,
-            DECRQSS,
-            DECSIXEL,
-            XTGETTCAP,
-
-            // OSC
-            SETICON,
-            SETTITLE,
-            SETWINTITLE,
-            SETXPROP,
-            SETCOLPAL,
-            SETCWD,
-            HYPERLINK,
-            COLORFG,
-            COLORBG,
-            COLORCURSOR,
-            COLORMOUSEFG,
-            COLORMOUSEBG,
-            SETFONT,
-            SETFONTALL,
-            CLIPBOARD,
-            RCOLPAL,
-            COLORSPECIAL,
-            RCOLORFG,
-            RCOLORBG,
-            RCOLORCURSOR,
-            RCOLORMOUSEFG,
-            RCOLORMOUSEBG,
-            RCOLORHIGHLIGHTFG,
-            RCOLORHIGHLIGHTBG,
-            NOTIFY,
-            DUMPSTATE,
-        };
-        crispy::sort(f, [](FunctionDefinition const& a, FunctionDefinition const& b) constexpr {
+    static auto const funcs = []() constexpr {
+        auto funcs = allFunctionsArray();
+        crispy::sort(funcs, [](FunctionDefinition const& a, FunctionDefinition const& b) constexpr {
```

Comment:
this is the workaround you're referring to? One function for declaring the array and this to make it sorted?

Reply:
Yes, Since gcc 9.4 doesn't have std::algorithms we can't call crispy::sort in constexpr contexts, becuase it uses std::swap. So instead of doing constexpr re-implementation of std::swap we can separate sorting and actual array temporarily until we move away from gcc 9.4.